tests/unit/tests_standard_lib/test_cim_tests/test_field_test_helper.py

```diff
@@ -0,0 +1,311 @@
+                ),
+            ],
+            "index=* event=alert OR event=emergency"
+            "\n| eval alert_valid=alert"
+            "\n| eval emergency_valid=emergency"
+            " \n| stats count as event_count"
+            ", count(alert) as alert_count"
+            ", count(emergency) as emergency_count"
+            " by sourcetype, source",
+        ),
+        (
+            None,
+            "index=* event=alert OR event=emergency"
+            " \n| stats count as event_count"
+            " by sourcetype, source",
+        ),
+    ],
+)
+def test_make_search_query(mocked_field_test_helper, fields, expected_output):
+    mocked_field_test_helper.fields = fields or []
+    with patch.object(
+        FieldTestHelper, "_gen_condition", return_value="event=alert OR event=emergency"
+    ):
+        mocked_field_test_helper._make_search_query("index=*")
+    assert (
+        mocked_field_test_helper.search_event
+        == "index=* event=alert OR event=emergency"
+    )
+    assert mocked_field_test_helper.search == expected_output
+
+
+@pytest.mark.parametrize(
+    "fields, expected_output",
+    [
+        (
+            None,
+            "Source, Sourcetype, Event Count\nutility.log, splunkd, 12\nsys.log, sc4s, 69\n\nSearch = base_search",
+        ),
+        (
+            [
+                field(name="field_1", get_properties=lambda: "field_1_properties"),
+                field(
+                    name="unknown_field",
+                    get_properties=lambda: "unknown_field_properties",
+                ),
+            ],
+            "Source, Sourcetype, Field, Event Count, Field Count, Invalid Field Count, Invalid Values\n"
+            "utility.log, splunkd, field_1, 12, 10, 0, -\n"
+            "sys.log, sc4s, unknown_field, 69, 9, 8, ['null', 'none']"
+            "\n\nSearch = base_search"
+            "\n\nProperties for the field :: field_1_properties"
+            "\n\nProperties for the field :: unknown_field_properties",
+        ),
+    ],
+)
+def test_format_exc_message(mocked_field_test_helper, fields, expected_output):
+    mocked_field_test_helper.parsed_result = [
+        {
+            "sourcetype": "splunkd",
+            "source": "utility.log",
+            "field": field(name="field_1"),
+            "event_count": 12,
+            "field_count": 10,
+            "valid_field_count": 10,
+            "invalid_values": [],
+        },
+        {
+            "sourcetype": "sc4s",
+            "source": "sys.log",
+            "field": field(name="unknown_field"),
+            "event_count": 69,
+            "field_count": 9,
+            "valid_field_count": 1,
+            "invalid_values": ["null", "none"],
+        },
+    ]
+    mocked_field_test_helper.fields = fields or []
+    mocked_field_test_helper.search = "base_search"
+    new_line = "\n"
+    with patch.object(
+        FieldTestHelper,
+        "get_table_output",
```

Comment:
that lambda is a bit complex and hard to understand. Maybe define normal function to make it more clear